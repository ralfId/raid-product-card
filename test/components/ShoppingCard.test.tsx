import React from 'react';
import renderer from 'react-test-renderer';
import { ShoppingCard } from '../../src/components';
import { product2 } from '../data/data';

const { act } = renderer;

describe('Shopping Card', () => {
    test('debe mostrar el componente correctamente', () => {
        const wrapper = renderer.create(
            <ShoppingCard product={product2}>
                {
                    () => (
                        <h2>Product Card</h2>
                    )
                }
            </ShoppingCard>);

        //console.log(wrapper.toJSON());
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})

describe('Shopping Card ', () => {
    test('debe incrementar el contador', () => {
        const wrapper = renderer.create(
            <ShoppingCard product={product2}>
                {
                    ({ count, increasBy }) => (

                        <>
                            <h2>Product Card</h2>
                            <span>{count}</span>
                            <button onClick={() => { increasBy(1) }}></button>
                        </>

                    )
                }
            </ShoppingCard>
        );

        //console.log(wrapper.toJSON());
        //expect(wrapper.toJSON()).toMatchSnapshot();

        let tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();

        act(() => {

            (tree as any).children[2].props.onClick();
        });

        tree = wrapper.toJSON();

        expect((tree as any).children[1].children[0]).toBe('1');
    })
})