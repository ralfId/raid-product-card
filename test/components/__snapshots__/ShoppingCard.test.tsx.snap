// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shopping Card  debe incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h2>
    Product Card
  </h2>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`Shopping Card debe mostrar el componente correctamente 1`] = `
<div
  className="productCard undefined"
>
  <h2>
    Product Card
  </h2>
</div>
`;
